mcfly.fish

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env fish
+
+# Ensure stdin is a tty
+test -t 0; or exit 0
+
+# Avoid loading this file more than once
+if test "$__MCFLY_LOADED" = "loaded"
+  exit 0
+end
+set -g __MCFLY_LOADED "loaded"
+
+# Note: we only use the history file for the session when this file was sourced.
+# Would have to reset this before calling mcfly if you want commands from another session later.
+if not set -q HISTFILE
+  set -gx HISTFILE (set -q XDG_DATA_HOME; and echo $XDG_DATA_HOME; or echo $HOME/.local/share)/fish/(set -q fish_history; and echo $fish_history; or echo fish)_history
```

Comment:
Yeah, this is the documented logic for finding the history file.  `XDG_DATA_HOME` is the preferred directory prefix if set, and `fish_history` is used as the session name if it's set, otherwise it falls back to `~/.local/share` and `fish`, respectively.

https://fishshell.com/docs/current/cmds/history.html#customizing-the-name-of-the-history-file
https://fishshell.com/docs/current/index.html#history-search